lib/src/json_schema/json_schema.dart

```diff
@@ -1934,6 +1934,35 @@ class JsonSchema {
         }
       });
 
+  _setDependentSchemas(dynamic value) => (TypeValidators.object('dependentSchemas', value)).forEach((k, v) {
+        if (v is Map || v is bool && schemaVersion >= SchemaVersion.draft6) {
+          _createOrRetrieveSchema('$_path/dependentSchemas/$k', v, (rhs) => _schemaDependencies[k] = rhs);
+        } else {
+          throw FormatExceptions.error('dependentSchemas values must be object (or boolean in draft6 and later): $v');
+        }
+      });
+
+  _setDependentRequired(dynamic value) => (TypeValidators.object('dependentRequired', value)).forEach((k, v) {
+        if (v is List) {
+          // Dependencies must have contents in draft4, but can be empty in draft6 and later
+          if (schemaVersion == SchemaVersion.draft4) {
+            if (v.isEmpty) throw FormatExceptions.error('dependentRequired item must be non-empty array');
```

Comment:
Sorry I probably wasn't clear. I was suggesting removing `item` from this message.